### PR TITLE
merging clickable image and 2d projection examples

### PR DESCRIPTION
**Pull request by choldgraf**
this converts the clickableimage example into a more general example showing how to convert 3D representations of electrodes into 2D, either using the mayavi 2D projection, or manually clicking w/ clickableimage

### CODE REVIEW

File: mne/viz/utils.py [hunk 1]
```diff
@@ -1031,7 +1031,7 @@ def __init__(self, imdata, **kwargs):  # noqa: D102
         self.ax = self.fig.add_subplot(111)
         self.ymax = self.imdata.shape[0]
         self.xmax = self.imdata.shape[1]
-        self.im = self.ax.imshow(imdata, aspect='auto',
+        self.im = self.ax.imshow(imdata,
```
Comment (Eric89GXL) [inline mne/viz/utils.py]: maybe it should be `aspect='equal'`? Or is that the default?

File: examples/visualization/plot_3d_to_2d.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+====================================================
+How to convert 3D electrode positions to a 2D image.
+====================================================
+
+Sometimes we want to convert a 3D representation of electrodes into a 2D
+image. For example, if we are using electrocorticography it is common to
+create scatterplots on top of a brain, with each point representing an
+electrode.
+
+In this example, we'll show two ways of doing this in MNE-python. First,
```
Comment (Eric89GXL) [inline examples/visualization/plot_3d_to_2d.py]: MNE-Python

File: examples/visualization/plot_3d_to_2d.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+if we have the 3D locations of each electrode then we can use Mayavi to
+take a snapshot of a view of the brain. If we do not have these 3D locations,
+and only have a 2D image of the electrodes on the brain, we can use the
+`mne.viz.ClickableImage` class to choose our own electrode positions
```
Comment (Eric89GXL) [inline examples/visualization/plot_3d_to_2d.py]: `:class:`

**Comment by choldgraf** [comment]
mmk added reference and made changes etc

File: examples/visualization/plot_3d_to_2d.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+# positions and turn them into a plottable layout.
+
+
+# This code opens the image so you can click on it. Commented out
+# because we've stored the clicks as a layout file already.
+
+# # The click coordinates are stored as a list of tuples
+# im = plt.imread('./brain.png')
+# click = ClickableImage(im)
+# click.plot_clicks()
```
Comment (agramfort) [inline examples/visualization/plot_3d_to_2d.py]: I tried to comment out

im = plt.imread('./brain.png')
click = ClickableImage(im)
click.plot_clicks()

but I get 

ValueError: not enough values to unpack (expected 2, got 0)

I need I demo I guess...

File: examples/visualization/plot_3d_to_2d.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+# coords = click.coords
+
+# # Generate a layout from our clicks and normalize by the image
+# print('Generating and saving layout...')
+# lt = generate_2d_layout(np.vstack(coords), bg_image=im)
```
Comment (agramfort) [inline examples/visualization/plot_3d_to_2d.py]: why the vstack? why not make coord already of the right shape?

File: examples/visualization/plot_3d_to_2d.py (hunk 4 above)
Comment (choldgraf) [inline examples/visualization/plot_3d_to_2d.py]: did you not click anywhere on the image? sounds like that might have been the issue. Either way the latest commit adds a check to make sure `self.coords` has something in it

**Comment by choldgraf** [comment]
any idea why circle just failed?

**Comment by Eric89GXL** [comment]
No idea, just restarted it.

**Comment by agramfort** [comment]
thx @choldgraf

**Pull request merged**